Library/include/CSP/Common/Map.h

```diff
@@ -81,19 +81,26 @@ template <typename TKey, typename TValue> class CSP_API Map
 		csp::memory::DllFree(Container);
 	}
 
-	/// @brief Returns an element at the given index of the map.
-	/// @param Key const TKey& : Elements key to access
+	/// @brief Returns a reference to the element with the given key in this map.
+	///        Will create a new element if the given key is not present.
+	/// @param Key const TKey& : Key of element in this map
 	/// @return TValue& : Map element
 	TValue& operator[](const TKey& Key)
 	{
 		return Container->operator[](Key);
 	}
 
-	/// @brief Returns a const element at the given index of the map.
-	/// @param Key const TKey& : Elements key to access
+	/// @brief Returns a const reference to the element with the given key in this map.
+	///        Throws if the given key is not present.
+	/// @param Key const TKey& : Key of element in this map
 	/// @return const TValue& : Map element
 	const TValue& operator[](const TKey& Key) const
 	{
+		if (Container->count(Key) == 0)
+		{
+			throw std::runtime_error("Key not present in Map. Please ");
```

Comment:
Looks like there's some missing text after 'Please '?

Reply:
Perhaps the second sentence could read. "Please validate the key first exists using `HasKey` ".

Reply:
I have no idea how this happened. I distinctly remember completing that sentence. Will fix now.